test/module/irohad/torii/torii_transport_command_test.cpp

```diff
@@ -0,0 +1,301 @@
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Return(shared_model::validation::Answer{}));
+  EXPECT_CALL(
+      *batch_factory,
+      createTransactionBatch(
+          A<const shared_model::interface::types::SharedTxsCollectionType &>()))
+      .Times(kTimes);
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(kTimes);
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service and number of invalid transactions
+ * @when calling ListTorii
+ * @then ensure that CommandService haven't called handleTransactionBatch
+ *       and StatusBus update status tx num times
+ */
+TEST_F(ToriiTransportCommandTest, ListToriiInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request;
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  shared_model::validation::Answer error;
+  error.addReason(std::make_pair("some error", std::vector<std::string>{}));
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Return(error));
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(0);
+  EXPECT_CALL(*status_bus, publish(_)).Times(kTimes);
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service
+ *        and some number of valid transactions
+ *        and one stateless invalid tx
+ * @when calling ListTorii
+ * @then ensure that CommandService haven't called handleTransactionBatch
+ *       and StatusBus publishes statelessInvalid for all txes
+ */
+TEST_F(ToriiTransportCommandTest, ListToriiPartialInvalid) {
+  grpc::ServerContext context;
+  google::protobuf::Empty response;
+
+  iroha::protocol::TxList request{};
+  for (size_t i = 0; i < kTimes; ++i) {
+    request.add_transactions();
+  }
+
+  int counter = 0;
+  EXPECT_CALL(*tx_validator, validate(_))
+      .Times(kTimes)
+      .WillRepeatedly(Invoke([&counter](const auto &) mutable {
+        shared_model::validation::Answer res;
+        if (counter++ == kTimes - 1) {
+          res.addReason(
+              std::make_pair("some error", std::vector<std::string>{}));
+        }
+        return res;
+      }));
+  EXPECT_CALL(
+      *batch_factory,
+      createTransactionBatch(
+          A<const shared_model::interface::types::SharedTxsCollectionType &>()))
+      .Times(kTimes - 1);
+
+  EXPECT_CALL(*command_service, handleTransactionBatch(_)).Times(kTimes - 1);
+  EXPECT_CALL(*status_bus, publish(_)).WillOnce(Invoke([](auto status) {
+    EXPECT_FALSE(status->statelessErrorOrCommandName().empty());
+  }));
+
+  transport_grpc->ListTorii(&context, &request, &response);
+}
+
+/**
+ * @given torii service and command_service with empty status stream
+ * @when calling StatusStream on transport
+ * @then Ok status is eventually returned without any fault
```

Comment:
Maybe add more assertions, such as "nothing is written to the stream"? Otherwise there is not much use of such test